src/test/java/kr/hhplus/be/server/order/OrderTest.java

```diff
@@ -40,6 +45,9 @@ public class OrderTest {
 
     @Mock
     private UserService userService;
+
+    @Mock
+    private ApplicationEventPublisher applicationEventPublisher;
```

Comment:
```suggestion
    @MockBean
    private ApplicationEventPublisher applicationEventPublisher;
```
@Mock은 단순 Mockito Mock 객체를 생성하지만 Spring Context에 등록되지 않아서 @Autowired나 의존성 주입 대상에 적용되지 않습니다.
@MockBean은 Spring 테스트 컨텍스트에 해당 Mock을 주입해 실제 빈을 대체하므로 통합 테스트에서 의존성 주입이 제대로 작동하려면 @MockBean을 사용해야 합니다.